enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -289,6 +289,76 @@ func uploadFiles(uploader *cachetools.BatchCASUploader, fc interfaces.FileCache,
 	return nil
 }
 
+// handleSymlink adds the symlink to the directory and actionResult proto so that
+// they could be recreated on the Bazel client side if needed.
+func handleSymlink(dirHelper *DirHelper, rootDir string, cmd *repb.Command, actionResult *repb.ActionResult, directory *repb.Directory, fqfn string) error {
+	target, err := os.Readlink(fqfn)
+	if err != nil {
+		return err
+	}
+	symlink := &repb.OutputSymlink{
+		Path:   trimPathPrefix(fqfn, rootDir),
+		Target: target,
+	}
+	directory.Symlinks = append(directory.Symlinks, &repb.SymlinkNode{
+		Name:   symlink.Path,
+		Target: symlink.Target,
+	})
+
+	// REAPI specification:
+	//   `output_symlinks` will only be populated if the command `output_paths` field
+	//   was used, and not the pre v2.1 `output_files` or `output_directories` fields.
+	//
+	// Check whether the current client is using REAPI version before or after v2.1.
+	if len(cmd.OutputPaths) > 0 && len(cmd.OutputFiles) == 0 && len(cmd.OutputDirectories) == 0 {
+		// REAPI >= v2.1
+		if dirHelper.IsOutputPath(fqfn) {
```

Comment:
You can flip this to `!dirHelper.IsOutputPath(fqfn)` and perform an early return.